app/assets/javascripts/lit/sources.js.coffee

```diff
@@ -0,0 +1,26 @@
+$ ->
+  sourceId = $('#source_id').attr('value')
+  apiKey = $('#api_key').attr('value')
+
+  updateFunc = ->
+    $.ajax "/lit/api/v1/sources/" + sourceId + "/sync_complete",
+      type: 'GET'
+      headers:
+        'Authorization': "Token token=\"#{apiKey}\""
+      success: (xml, textStatus, xhr) ->
+        if xhr.responseJSON.sync_complete
+          $('.loading').addClass('loaded').removeClass('loading')
+          clearInterval(interval)
+          location.reload()
+      statusCode:
+        404: ->
+          $('.loading').text('Could not update synchronization status, please try refreshing page')
+        401: ->
+          $('.loading').text('Invalid API key, please check source settings')
```

Comment:
The code is updated, but I think my first comment still applies here. I don't think 401 status can be returned here (although I'm not completely sure). And if it can, api key wan't be a reason, so at least message should be changed.